CHANGELOG.md

```diff
@@ -14,10 +14,24 @@ This project adheres to [Semantic Versioning](http://semver.org/).
     - new `contains` relational operator for strings
     - new readonly `derived_filter_structure` prop exposing the query structure in a programmatically friendlier way
 
+[#412](https://github.com/plotly/dash-table/pull/412)
+- Add support for row IDs, based on the `'id'` attribute of each row of `data`
+    - IDs will not be displayed unless there is a column with `id='id'`
+    - `active_cell`, `start_cell`, `end_cell`, and items in `selected_cells` contain row and column IDs: All are now dicts  `{'row', 'column', 'row_id' and 'column_id'}` rather than arrays `[row, column]`.
+    - Added new props mirroring all existing row indices props:
+        - `selected_row_ids` mirrors `selected_rows`
+        - `derived_viewport_row_ids` mirrors `derived_viewport_indices`
+        - `derived_virtual_row_ids` mirrors `derived_virtual_indices`
+        - `derived_viewport_selected_row_ids` mirrors `derived_viewport_selected_rows`
+        - `derived_virtual_selected_row_ids` mirrors `derived_virtual_selected_rows`
+
 ### Changed
 [#397](https://github.com/plotly/dash-table/pull/397)
 - Rename `filtering_settings` to `filter`
```

Comment:
I realize this wasn't done in this PR, but should there be some symmetry with `sorting_settings` -> `sort` here?

Reply:
Yes. Used the filtering breaking changes as an opportunity to rename the props but limited myself to those.

Reply:
Now does seem like a good time to change this, but I might vote for `sort_by` - `sort` could be interpreted to mean "only sort these columns," rather than the actual "sort all columns based on these ones"